### ISSUE TITLE
[Tasks]: Explore generic_deltamodel

### ISSUE THREAD

**Issue by SarahAlidoost** [WUR-AI/diffWOFOST#7]
generic_deltamodel is a framework developed for the hydrological model HBV. See the resource:
- https://github.com/mhpi/generic_deltamodel 
- https://github.com/mhpi/dhbv2

We want to answer the questions at #2.

**Comment by luisaforozco** [comment]
Thanks @SarahAlidoost for this overview, indeed very nice observations, I would like to add:
- Use of of blocks `with torch.no_grad():` for code that cannot be differentiable (e.g. because using an external library, because the operation does not require to track gradients).

Also, I would like to highlight the difference between two approaches: _a-priori_ and _a_posteriori_ fitting. 
1. _A-priori_: the loss function calculates the difference between $P$ and labels. In this case $HBV$ does not need to be differentiable.
2. _A-posteriori_: the loss function calculates the difference between $Q, \mu$. In this case $HBV$ has to be differentiable.
